feat: Add custom Azure endpoints support and update charts to v0.14.2/2.14.2

Krishna-Chaitanya4 opened the pull request:
## 🚀 Changes

### Added
- **Custom Azure Endpoints Support**: Added configurable endpoints mapping for Azure landscape configuration
  - Allows users to specify custom Azure service endpoints through `azureLandscape.endpoints` values
  - Supports dynamic key-value pairs for different Azure services
  - Conditional rendering only when endpoints are configured

### Updated  
- **Chart Versions**: 
  - `promitor-agent-resource-discovery`: Updated to version `0.14.2`
  - `promitor-agent-scraper`: Updated to version `2.14.2`
- **Generated new chart packages** with updated versions

## 📝 Configuration

The new endpoints feature can be configured in your `values.yaml`:

```yaml
azureLandscape:
  endpoints:
    resourceManager: https://management.azure.com/
    activeDirectory: https://login.microsoftonline.com/
    # Add any additional custom endpoints as needed

Commit:
Added support for custom endpoints and updated chart versions to 0.14.2

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/promitor/charts?pullRequest=188) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/promitor/charts?pullRequest=188) before we can accept your contribution.<br/><hr/>**Krishna Chaitanya Yadav Perugu** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/promitor/charts?pullRequest=188) it.</sub>

Commit:
Added support for custom endpoints and updated chart versions to 0.14.2 with appversion 0.14.1

Commit:
Fix YAML parsing error in ServiceAccount templates

- Fixed annotations rendering in both promitor-agent-resource-discovery and promitor-agent-scraper ServiceAccount templates
- Changed from inline toYaml rendering to properly indented structure
- Resolves deployment error: 'cannot unmarshal object into Go struct field .metadata.annotations of type string'

Commit:
Revert "Fix YAML parsing error in ServiceAccount templates"

This reverts commit 5726919c5ec9c92c73a3391faf146bb076f7d491.

tomkerkhove reviewed (changes requested):
Added some comments, but please also update the README of the packages with the new options.

Thanks for opening the PR!
